tower-route/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+extern crate futures;
+extern crate tower_layer;
+extern crate tower_service;
+extern crate void;
+
+use futures::{future::Either, Async, Poll};
+use tower_service::Service;
+
+pub trait Acceptor<Request> {
+    fn accepts(&self, request: &Request) -> bool;
+}
+
+pub struct Route<A, B> {
+    service: A,
+    other: B,
+}
+
+impl<A, B, Request, Future, Response, Error> Service<Request> for Route<A, B>
```

Comment:
I don’t think you need most of these generics. You can say B: Service<Request, Reaponse = A::Response, ...>

You also don’t want the future types to be the same or each service won’t be able to have different behavior :-)